eng/build.sh

```diff
@@ -211,33 +205,29 @@ function BuildSolution {
     quiet_restore=true
   fi
 
-  # Node reuse fails because multiple different versions of FSharp.Build.dll get loaded into MSBuild nodes
-  node_reuse=false
-
   # build bootstrap tools
   bootstrap_config=Proto
+  bootstrap_tfm=netcoreapp2.1
+  MSBuild "$repo_root/src/buildtools/buildtools.proj" \
+    /restore \
+    /p:Configuration=$bootstrap_config \
+    /nodeReuse:false \
+    /t:Build
+
   bootstrap_dir=$artifacts_dir/Bootstrap
-  if [[ "$force_bootstrap" == true ]]; then
-     rm -fr $bootstrap_dir
-  fi
-  if [ ! -f "$bootstrap_dir/fslex.dll" ]; then
-    MSBuild "$repo_root/src/buildtools/buildtools.proj" \
-      /restore \
-      /p:Configuration=$bootstrap_config \
-      /t:Publish
+  mkdir -p "$bootstrap_dir"
+  cp $artifacts_dir/bin/fslex/$bootstrap_config/netcoreapp2.1/* $bootstrap_dir
+  cp $artifacts_dir/bin/fsyacc/$bootstrap_config/netcoreapp2.1/* $bootstrap_dir
 
-    mkdir -p "$bootstrap_dir"
-    cp -pr $artifacts_dir/bin/fslex/$bootstrap_config/netcoreapp2.1/publish $bootstrap_dir/fslex
-    cp -pr $artifacts_dir/bin/fsyacc/$bootstrap_config/netcoreapp2.1/publish $bootstrap_dir/fsyacc
-  fi
-  if [ ! -f "$bootstrap_dir/fsc.exe" ]; then
-    MSBuild "$repo_root/proto.proj" \
-      /restore \
-      /p:Configuration=$bootstrap_config \
-      /t:Publish
+  MSBuild "$repo_root/proto.proj" \
```

Comment:
Why are these build steps being done unconditionally?  Won't this rebuild the proto compiler on every change to the compiler source?

Reply:
They were added by your PR and rolled back by the reversion, I agree they are a good thing, and should be added back.